src/topp/FLASHDeconv.cpp

```diff
@@ -148,7 +148,7 @@ class TOPPFLASHDeconv : public TOPPBase
     setMinInt_("max_MS_level", 1);
 
     registerIntOption_("forced_MS_level", "", 0,
-                       "If set to an integer N, MS level of all spectra will be set to N regardless of original MS level. Useful when deconvolving datasets containing only MS2 spectra.", false, true);
+                       "If set to an integer N, also deconvolute all spectra of level N and set precursor charge to mzml_mass_charge. Useful when deconvolving datasets containing only MS2 spectra.", false, true);
```

Comment:
@poshul  : I am not sure if this parameter was intended for this use.  I introduced this parameter for many native MS datasets where only MS2 spectra are generated due to fixed isolation windows. To forcefully change MS levels into 1 (i.e., N is set to 1), it was easier to visualize the deconvolved masses in, e.g., Toppview. But maybe I misunderstood your change.. Could you add a bit of example on how your suggested change will affect analysis?

Reply:
The use case that this came up for for me was when we had MS2 scans only from a direct infusion experiment with a single Oligonucleotide. Before the changes in this PR, the output from FLASHDeconv listed all those spectra as MS1 level, and didn't update the precursor. I handled this by editing the output MzML itself. These changes update the MS level of the deconvolved peaks back to that of the input (ie MS level 2), and updates the precursor mass and charge to match the deconvolution. Looking back at this there is still potentially weird behavior where the input MzML had multiple levels (MS1 and MS2), I can try to make changes to make sure the correct MS level is maintained if you think this PR is overall going to be useful.

Reply:
> 2. If this parameter "forced_MS_level" is used, it forcefully change the ms level into the specified one. Well, or maybe we do not need this functionality any more.. This is just for visualization....  Does Toppview have an option to visualize MS2 level heatmap?

@cbielow how difficult would it be to enable plotting MS2 as heat map? I think this could also be an interesting visualization for SWATH data

Reply:
not sure what is meant by "heat map", but if that means the usual 2D plot which we have for MS1 data, then its mostly a matter of configuration. In some places there are hardcoded checks for MS level which would require parametrization.

Reply:
"Matter of configuration" meaning the heat map (yes, the x-axis retention time y-axis m/z and z-axis intensity color coded) can be shown for MS2 spectra just by selecting options available in toppview? Or should there be change in the source code?

Reply:
I think it would need some changes in TOPPView to display different levels and a way to switch between MS1 and MS2...